lib/sort_by_length.rb

```diff
@@ -1,7 +1,35 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2) insertion sort
+# Space complexity: 0(n) created new array&hash that have the linear relationship with the length of the input
 def sort_by_length(my_sentence)
-  raise NotImplementedError, "Method not implemented"
+  my_sentence = my_sentence.split(" ")
+  sentence_hash = Hash.new
+  my_sentence.map do |word|
+    sentence_hash[word] = word.length
+  end
+
+  array = sentence_hash.values
+  i = 1
+  while i < my_sentence.length
+    value_to_insert = array[i]
+    j = i
+
+    while j > 0 && array[j-1] > value_to_insert
+      array[j] = array[j-1]
+      j -= 1
+    end
+    array[j] = value_to_insert
+    i += 1
+  end
+
+  word = []
+  array.map do |length|
+    word << sentence_hash.key(length)
```

Comment:
Just note that `sentence_hash.key` is an O(n) operation because it has to loop through the hash to find a key with that matching value.